### PR TITLE
feat: load updates from mongo instead of api.oboapp.online

### PR DESCRIPTION
**Pull request by vbuch**
# Pull Request

## Description

Switch updates loading from api.oboapp.online to a local MongoDB.

## Related Issue

Closes #

## Changes Made

- Added `GET /api/updates-export` with API-key auth and a configurable maximum result size.
- Refactored `GET /api/updates` and `GET /api/updates/by-id` to query MongoDB directly using shared mapping utilities.
- Updated OpenAPI schema, tests, and environment examples to reflect the new endpoints and configuration.
- Updated sendUpdatesNotifications to also use Mongo.

## Screenshots/Videos (if applicable)

Screenshot_20260522_003503

## Testing

- Set up an instance of OboApp locally working with MongoDB
- Configure ysm-api (.env) to connect to that same MongoDB
- Run a crawl and ingest in the OboApp instance to populate the `messages` collection in Mongo
- Verify that YSM shows the available messages

### CODE REVIEW

File: src/lib/oboMongo.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return Promise.reject(new Error('YSM_OBOAPP_MONGODB_URI is not configured'))
+  }
+  const client = new MongoClient(uri)
+  return client.connect()
+}
+
+function getClientPromise(): Promise<MongoClient> {
+  if (process.env.NODE_ENV === 'development') {
+    if (!global.__ysm_obo_mongo_client_promise) {
+      global.__ysm_obo_mongo_client_promise = createClientPromise()
```
[the remark this answers is not in the document]
Comment (vbuch) [reply src/lib/oboMongo.ts]: Addressed.

File: src/endpoints/updates.ts [hunk 2]
```diff
@@ -198,45 +188,124 @@ export const updates: Endpoint = {
...
+    }
+    const north = toNum(rawQuery.north)
+    const south = toNum(rawQuery.south)
+    const east = toNum(rawQuery.east)
+    const west = toNum(rawQuery.west)
+    const hasBounds =
+      north !== undefined && south !== undefined && east !== undefined && west !== undefined
+    const bounds: ViewportBounds | null = hasBounds
+      ? { north: north!, south: south!, east: east!, west: west! }
+      : null
```
[the remark this answers is not in the document]
Comment (vbuch) [reply src/endpoints/updates.ts:205]: Fixed —  now returns  for empty/whitespace strings before calling , so  will correctly be treated as missing bounds.

File: src/endpoints/updatesExport.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    if (rawMessages.length > limitMax) {
+      const messageCount = await messagesCollection.countDocuments(messageFilter)
+      return Response.json(
+        {
+          error: 'limitExceeded',
+          messageCount,
+        },
+        { status: 413 }
+      )
+    }
```
[the remark this answers is not in the document]
Comment (vbuch) [reply src/endpoints/updatesExport.ts:173]: Fixed — wrapped the `countDocuments` call in a try/catch that logs and returns a controlled 500 on failure.

File: src/endpoints/updatesExport.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    // ── Auth ──────────────────────────────────────────────────────────────
+    const expectedKey = process.env.YSM_OBOAPP_SYNC_API_KEY
+    if (!expectedKey) {
+      return Response.json({ error: 'YSM_OBOAPP_SYNC_API_KEY is not configured' }, { status: 500 })
+    }
+
+    const providedKey = req.headers.get('x-api-key')
+    if (!providedKey || providedKey !== expectedKey) {
+      return Response.json({ error: 'Unauthorized' }, { status: 401 })
+    }
```
[the remark this answers is not in the document]
Comment (vbuch) [reply src/endpoints/updatesExport.ts:102]: Fixed — switched to `crypto.timingSafeEqual` with a zero-filled fixed-length buffer so the comparison always runs in constant time, then verifies lengths match separately to prevent length oracle leakage.

File: src/lib/oboMessageMapper.ts [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+  try {
+    const createdAt = toRequiredISOString(doc.createdAt, new Date().toISOString())
+    const { timespanStart, timespanEnd } = resolveTimespan(doc, createdAt)
+
+    const id =
+      typeof doc._id === 'string'
+        ? doc._id
+        : doc._id && typeof (doc._id as Record<string, unknown>).toString === 'function'
+          ? String(doc._id)
+          : undefined
```
[the remark this answers is not in the document]
Comment (vbuch) [reply src/lib/oboMessageMapper.ts:279]: Added `src/lib/__tests__/oboMessageMapper.test.ts` with 48 tests covering all the mentioned edge cases: `geoJson` as stringified JSON, Firestore `toDate()` shape, malformed doc returning `null`, all four timespan fallback rules, plus `hasNullPinTimespanStart`, `messageInBounds` (cityWide, Point/LineString/Polygon centroids, pin coordinates), and `sortByRelevance`.
